_posts/2018-10-08-Pushing-Images-to-Azure.md

```diff
@@ -0,0 +1,144 @@
+---
+
+layout: post
+title: "Pushing composed images to Azure"
+author: swalter
+tags: welder-web lorax-composer
+
+---
+
+Weldr aka. Composer can generate images suitable for uploading to the Azure
+cloud, and running an instance there. The images have the right format, and
+include the necessary agents, as well as
+[cloud-init](https://cloudinit.readthedocs.io/en/latest/index.html).
+
+## Prerequisites
+
+We'll use [Fedora 29](https://getfedora.org/) as our OS of choice for running this. Run
+this in its own VM with at least 8 gigabytes of memory and 40 gigabytes of disk space.
+[Lorax](http://weldr.io/lorax/) makes some changes to the operating system its running on.
+
+First install Composer:
+
+    $ sudo yum install lorax-composer cockpit-composer cockpit composer-cli
+
+Next make sure to turn off [SELinux](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/5/html/deployment_guide/ch-selinux) on the system. Lorax doesn't yet work properly with
+SELinux running, as it installs an entire OS image in an alternate directory:
+
+    $ sudo setenforce 0
+    $ sudo sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/selinux/config
+
+Now start ```lorax-composer``` system service:
+
+    $ sudo systemctl enable --now lorax-composer
+
+If you're going to use [Cockpit](https://cockpit-project.org/) UI to drive Composer
+(see below), you can also enable it like this:
+
+    $ sudo systemctl enable --now cockpit.socket
+    $ sudo firewall-cmd --add-service=cockpit && firewall-cmd --add-service=cockpit --permanent
+
+Install the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-yum) tooling:
+
+    $ sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+    $ sudo sh -c 'echo -e "[azure-cli]\nname=Azure CLI\nbaseurl=https://packages.microsoft.com/yumrepos/azure-cli\nenabled=1\ngpgcheck=1\ngpgkey=https://packages.microsoft.com/keys/microsoft.asc" > /etc/yum.repos.d/azure-cli.repo'
+    $ sudo yum install azure-cli
```

Comment:
You are using F29, but say yum here instead of dnf.

Reply:
I'd like the documentation to work for RHEL as much as possible. Yum works on both.